src/Compilers/CSharp/Test/Emit3/Semantics/DictionaryExpressionTests.cs

```diff
@@ -0,0 +1,616 @@
+                // d = [1:"one"];
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"[1:""one""]").WithArguments("System.Collections.Generic.Dictionary`2", ".ctor").WithLocation(4, 5),
+                // (5,5): error CS0656: Missing compiler required member 'System.Collections.Generic.Dictionary`2..ctor'
+                // d = [new KeyValuePair<int, string>(2, "two")];
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"[new KeyValuePair<int, string>(2, ""two"")]").WithArguments("System.Collections.Generic.Dictionary`2", ".ctor").WithLocation(5, 5),
+                // (6,5): error CS0656: Missing compiler required member 'System.Collections.Generic.Dictionary`2..ctor'
+                // d = [.. new KeyValuePair<int, string>[] { new(3, "three") }];
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"[.. new KeyValuePair<int, string>[] { new(3, ""three"") }]").WithArguments("System.Collections.Generic.Dictionary`2", ".ctor").WithLocation(6, 5));
+
+            comp = CreateCompilation(source);
+            comp.MakeMemberMissing(WellKnownMember.System_Collections_Generic_Dictionary_KV__Add);
+            comp.VerifyEmitDiagnostics(
+                // (3,5): error CS0656: Missing compiler required member 'System.Collections.Generic.Dictionary`2.Add'
+                // d = [];
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "[]").WithArguments("System.Collections.Generic.Dictionary`2", "Add").WithLocation(3, 5),
+                // (4,5): error CS0656: Missing compiler required member 'System.Collections.Generic.Dictionary`2.Add'
+                // d = [1:"one"];
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"[1:""one""]").WithArguments("System.Collections.Generic.Dictionary`2", "Add").WithLocation(4, 5),
+                // (5,5): error CS0656: Missing compiler required member 'System.Collections.Generic.Dictionary`2.Add'
+                // d = [new KeyValuePair<int, string>(2, "two")];
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"[new KeyValuePair<int, string>(2, ""two"")]").WithArguments("System.Collections.Generic.Dictionary`2", "Add").WithLocation(5, 5),
+                // (6,5): error CS0656: Missing compiler required member 'System.Collections.Generic.Dictionary`2.Add'
+                // d = [.. new KeyValuePair<int, string>[] { new(3, "three") }];
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, @"[.. new KeyValuePair<int, string>[] { new(3, ""three"") }]").WithArguments("System.Collections.Generic.Dictionary`2", "Add").WithLocation(6, 5));
+        }
+
+        [Theory]
+        [InlineData("IDictionary<int, string>")]
+        [InlineData("IReadOnlyDictionary<int, string>")]
+        public void KeyValuePair_MissingMember(string typeName)
+        {
+            string source = $$"""
+                using System.Collections.Generic;
+                {{typeName}} d;
+                d = [];
```

Comment:
Do we have a successful test with empty collection expression?

Reply:
Was this resolved?

Reply:
Yes, added test `DictionaryInterface_01`.

Reply:
Ah, I searched for `= []` so missed it. Thanks